pilot/pkg/serviceregistry/kube/controller/ambient/services_test.go

```diff
@@ -1120,9 +1124,99 @@ func TestServiceConditions(t *testing.T) {
 			builder := a.serviceServiceBuilder(
 				krttest.GetMockCollection[Waypoint](mock),
 				krttest.GetMockCollection[*v1.Namespace](mock),
+				krttest.GetMockSingleton[MeshConfig](mock),
 			)
 			res := builder(krt.TestingDummyContext{}, tt.svc)
 			assert.Equal(t, res.GetConditions(), tt.conditions)
 		})
 	}
 }
+
+func TestMatchServiceScope(t *testing.T) {
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ns",
+			Labels: map[string]string{
+				v1.LabelMetadataName: "ns",
+				"istio.io/dataplane-mode": "ambient",
+			},
+		},
+	}
+	
+	serviceOptInMeshConfig := mesh.DefaultMeshConfig()
+	serviceOptInMeshConfig.ServiceScopeConfigs = []*meshConfig.MeshConfig_ServiceScopeConfigs{
+		{
+			ServicesSelector: &meshConfig.LabelSelector{
+				MatchExpressions: []*meshConfig.LabelSelectorRequirement{
+					{
+						Key:      "istio.io/global",
+						Operator: string(metav1.LabelSelectorOpExists),
+					},
+				},
+			},
+			Scope: meshConfig.MeshConfig_ServiceScopeConfigs_GLOBAL,
+		},
+	}
+	serviceOptInMeshConfigResource := &meshwatcher.MeshConfigResource{
+		MeshConfig: serviceOptInMeshConfig,
+	}
+
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "svc",
+			Namespace: "ns",
+			Labels: map[string]string{
+				v1.LabelMetadataName: "svc",
+				"istio.io/global":    "true",
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Type: v1.ServiceTypeClusterIP,
+		},
+	}
+
+	tests := []struct {
+		name string
+		inputs     []any
+		meshCfg *MeshConfig
+		svc 	  *v1.Service
+		want model.ServiceScope
+	}{
+		{
```

Comment:
nit: we probably want more test cases for the final PR to make sure every permutation is caught. One of the comments in the API PR had a good list